### DISCUSSION TITLE
Does the Path Tracer activly searches for a Lightsource after beeing reflected at a Surface

### DISCUSSION THREAD

**Discussion by Tscho1995** [mitsuba-renderer/mitsuba3#514] [Q&A] [answered]
Hello,

in our Research we implement structed Ligth emitters to Mitsuba. As a result, the Emitter can be very small. Hence it can be hard to find a path from Camera to Light source. Now in the Documentation for Path Integrator is written "At every intersection, the path tracer tries to create a connection to the light source in an attempt to find a complete path along which light can flow from the emitter to the sensor. "

Does that mean, that the rays activly try to find the light source in the Scene? Or is it all just based on the reflection Angle of the Surface?

Thank You

**Comment by ziyi-zhang** [comment, marked as answer]
Hi,
Source code of path integrator with comments can be found [here](https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/integrators/path.cpp).

What you described "find the light source" is the strategy `emitter sampling` in the code, and "reflection Angle of the Surface" is the strategy `BSDF sampling`. They are combined with `multiple importance sampling` (MIS) to get the same correct result with lower variance. The behavior to actively find emitters is also called `next event estimation`.
